cmake/mcuboot.cmake

```diff
@@ -158,6 +171,29 @@ function(zephyr_mcuboot_tasks)
       ${west_sign} ${encrypted_args} ${imgtool_args} --encrypt "${keyfile_enc}")
   endif()
   set_property(GLOBAL APPEND PROPERTY extra_post_build_byproducts ${byproducts})
+
+  # Regenerate the binary files from the hex files so that the content of both is
+  # the same. Otherwise the .hex and .bin will have different signatures due to the
+  # independent calls to the signing tool. This cannot be done as part of 'west sign'
+  # as we cannot evaluate the bintools arguments at configuration time.
+  if (CONFIG_MCUBOOT_GENERATED_HEX_BIN_SAME_CONTENT)
```

Comment:
instead of introducing yet another Kconfig setting, then why no swap the code here:
https://github.com/zephyrproject-rtos/zephyr/blob/ccd88a4b611ca4dfe3985709d3cd36021910a6c2/cmake/mcuboot.cmake#L110
and here:
https://github.com/zephyrproject-rtos/zephyr/blob/ccd88a4b611ca4dfe3985709d3cd36021910a6c2/cmake/mcuboot.cmake#L127

so that order becomes:
```
  if(CONFIG_BUILD_OUTPUT_HEX)
    # Do signing of hex file
  endif()

  if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_HEX)
    # Both hex and bin output is enabled, convert signed hex to signed bin to mitigate #52271.
  elseif(CONFIG_BUILD_OUTPUT_BIN)
    # Only bin output is enabled, keep existing image bin signing.
  endif()
```

Reply:
I went with this approach as `west sign` current automatically discovers files to be signed, so even without the `bin` options it still actually signs the `.bin`. If this file is generated anyway the easiest option IMO is just to overwrite it. Obviously this can be updated but is a larger change.

https://github.com/zephyrproject-rtos/zephyr/blob/fd94ca7e78e641462cd917dad197f655f76337be/scripts/west_commands/sign.py#L155-L175

I didn't go with `if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_HEX)` directly in the cmake just to provide the option to disable this behaviour if it happens to cause issues for someone.

Reply:
> I went with this approach as west sign current automatically discovers files to be signed

Right, I had forgotten about the pure `west sign` behavior.
Having different behavior on `west sign` and during build is not good.

> I didn't go with if(CONFIG_BUILD_OUTPUT_BIN AND CONFIG_BUILD_OUTPUT_HEX) directly in the cmake just to provide the option to disable this behaviour if it happens to cause issues for someone.

and that's a fair point.
But I dislike having an extra Kconfig for this.
and even though it's configurable, there is still a mismatch to how it works when `west sign` is invoked.